include/boost/url/const_string.hpp

```diff
@@ -0,0 +1,216 @@
+
+    However, care must be exercised; undefined
+    behavior results if the string_view
+    portion of the object is modified
+    directly, for example by calling
+    `remove_suffix` or `operator=`.
+
+    Slicing, however, is supported, as
+    copies of the `string_view` portion
+    of the object are valid and remain
+    valid for the lifetime of the original
+    object.
+*/
+class const_string : public string_view
+{
+    /** Type for const_string reference counter and custom destructor
+
+        This struct stores the number of references to this read-only string.
+        As the string is read-only, we can create a copy by simply updating
+        the reference counter.
+
+        As the underlying string is constructed with a custom allocator, the
+        struct also includes a custom destructor to deallocate the
+        string whenever the reference counter reaches zero.
+
+     */
+    struct base;
+
+    /** Instance of reference counter for @ref const_string instances pointing to this string
+
+        Pointer to the reference counter for all const_string instances
+        pointing to the same string.
+
+     */
+    base* p_ = nullptr;
+
+    /** Allocate a string copy and return its initial reference counter
+
+        This function allocates a string of input size `n` starting at
+        output address `dest` with the allocator `a`.
+
+        It returns its initial reference counter and the string address as
+        `dest`.
+
+        @note As an optimization, the initial reference counter is `nullptr` for
+        empty strings. For other strings, the reference counter starts as `1`
+        and contains a pointer to a `deallocate` function, which might be used
+        later on by @ref const_string to deallocate this string value.
+
+        @note If `n == 0`, `dest` is set to `nullptr` to logically represent
+        an empty string. This is unlike string_view, which differentiates
+        empty strings of size 0 and no string pointing to nullptr.
+
+        @tparam Allocator Allocator used to create allocate the new string
+
+        @return Pointer to the initial reference counter for this new string
+
+        @param n Size of the new string
+
+        @param a Allocator used to allocate the new string
+
+        @param dest Address where the string should be constructed
+
+
+     */
+    template<class Allocator>
+    base*
+    construct(
+        std::size_t n,
+        Allocator const& a,
+        char*& dest);
+
+public:
+    /** The default allocator type to construct read-only instances of @ref const_string
+
+        An allocator of this class can be instantiated to create
+        multiple read-only string values.
+
+        @note This custom allocator is not an instance of the C++ Allocator named requirement
+
+        @par References
+        @li <a href="https://en.cppreference.com/w/cpp/named_req/Allocator">Allocator (cppreference)</a>
+
+     */
+    class allocator;
```

Comment:
This can just be
```
#ifdef BOOST_URL_DOCS
using allocator = __see_below__;
#else
class allocator;
#endif

Reply:
OK. Removing the javadoc then.

Reply:
I think we might need something special for `allocator`